web/src/recent_view_ui.ts

```diff
@@ -483,7 +490,14 @@ export function process_messages(messages: Message[], msg_list_data?: MessageLis
 
     // Only rerender if conversation data actually changed.
     if (conversation_data_updated) {
-        complete_rerender();
+        if (!rows_order_changed) {
+            // If rows order didn't change, we can just rerender the affected rows.
+            for (const key of updated_rows) {
+                inplace_rerender(key);
```

Comment:
`inplace_rerender` doesn't seem designed to run in a loop; it calls `setTimeout(revive_current_focus, 0);` at the end.

It also looks like we resort `topics_widget` each time; could probably add a parameter that allows skipping that step.

Reply:
Okay, did both. Also, noticed we neet to call `topics_widget.replace_list_data(get_list_data_for_widget(), false);` if we are adding new rows. Add this comment for that too:

```
        // NOTE: This doesn't add any new entry to the original list but updates the filtered list
        // based on the current filters and updated row data.
        topics_widget.filter_and_sort();
```